autogen/agentchat/conversable_agent.py

```diff
@@ -1031,6 +1031,28 @@ def execute_code_blocks(self, code_blocks):
                     filename=filename,
                     **self._code_execution_config,
                 )
+            elif lang in ["js", "javascript", "JavaScript"]:
+                if code.startswith("# filename: "):
+                    filename = code[11 : code.find("\n")].strip()
+                else:
+                    filename = None
+                exitcode, logs, image = self.run_code(
+                    code,
+                    lang="javascript",
+                    filename=filename,
+                    **self._code_execution_config,
+                )
+            elif lang in ["ts", "typescript", "TypeScript"]:
+                if code.startswith("# filename: "):
+                    filename = code[11 : code.find("\n")].strip()
+                else:
+                    filename = None
+                exitcode, logs, image = self.run_code(
+                    code,
+                    lang="typescript",
+                    filename=filename,
+                    **self._code_execution_config,
+                )
```

Comment:
lots of common code shared w/ 1024-1033. Redundancy can be reduced.

Reply:
yes, i agree thanks!